MAUI/DataForm/DataForm-settings.md

```diff
@@ -0,0 +1,899 @@
+
+## Change the label name for auto-generated editor
+
+By default property name will be displayed as label text. You can change the label text by using display attribute or using the `GenerateDataFormItem` event when `AutoGenerate` is true.
+
+#### Using attribute
+
+You can change the label text by setting the `Name` or `ShortName` in display attribute.
+
+{% tabs %}
+{% highlight C# %}
+
+    
+    [Display(Name = "FirstName")]
+    public string Name { get; set; }
+    
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Using event
+
+You can change the label text by using the `LabelText` property in `DataFormItem` .
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "Name")
+            {
+                e.DataFormItem.LabelText = "FirstName";
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when `AutoGenerate` is true.
+
+For `AutoGenerate` false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor
+
+## Hide Label for an editor
+
+By default, the label will be generated by the property name. You can hide the label text by using the DataFormDisplayAttribute or using the `GenerateDataFormItem` event when `AutoGenerate` is true.
+
+For `AutoGenerate` false you can use property directly by using the DataFormItem - add refrence link for explict create data editor
+#### Using attribute
+
+You can hide the label text for editor by setting `ShowLabel` as false in `DataformDisplayOption` attribute.
+
+{% tabs %}
+{% highlight C# %}
+
+    
+    [DataFormDisplayOptions(ShowLabel = false)]
+    public string MiddleName { get; set; }
+    
+}
+{% endhighlight %}
+{% endtabs %}
+
+#### Using events
+
+You can hide the label text for editor by using the `ShowLabel` property in `DataFormItem`.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.GenerateDataFormItem += OnGenerateDataFormItem;
+
+    private void OnGenerateDataFormItem(object sender, GenerateDataFormItemEventArgs e)
+    {
+        if (e.DataFormItem != null)
+        {
+            if (e.DataFormItem.FieldName == "FirstName")
+            {
+                e.DataFormItem.ShowLabel = false;
+            }
+        }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+GenerateDataFormItem event when `AutoGenerate` is true.
+
+For `AutoGenerate` false you can use property directly by using the DataFormItem - add reference link for explicit creaate data editor
+
+## Change editor layout settings
+
+You can change the label width, label position and editor width. This is possible by using the `GenerateDataFormItem` event when `AutoGenerate` is true. In the event, change the layout by using the `DefaultLayoutSettings` property in the `DataFormItem`.
```

Comment:
You can change the label width, label position, and editor width. This is possible by using the `GenerateDataFormItem` event when `AutoGenerateItems` is true. In the event, change the layout settings by using the `DefaultLayoutSettings` property in the `DataFormItem`.

Reply:
content updated